Parse header md level

CorneliaKelinske opened the pull request:
I have simplified some stuff and added comments in the code regarding what I did.

I feel that in order to simplify even more, I would have to change the whole process flow of the original code.
It seems to me that in the original version the code is checked for headers, list, paras first and then for highlighting.
If I was to set this up, I would probably do the highlighting first and then process the rest.

What are your thoughts?
How much else do you think I should do?
where do I start?

alanvardy commented on elixir/markdown/lib/markdown.ex:
```diff
@@ -11,37 +11,53 @@ defmodule Markdown do
+    |> Enum.map(& process(&1))
+    |> IO.inspect(label: "Enum.map process output")
+    |> Enum.join()
+    |> IO.inspect(label: "output of Enum.join")
```
Do you want to remove the `IO.inspect`s? Using a tool like [Credo](https://github.com/rrrene/credo) can automate these checks for you.

alanvardy replied:
You can respond to the above comment or click "resolve conversation" to collapse this thread!

CorneliaKelinske replied:
There should be no more IO.inspects now

alanvardy replied:
:+1:

alanvardy commented on elixir/markdown/lib/markdown.ex:
```diff
@@ -11,37 +11,53 @@ defmodule Markdown do
+    |> IO.inspect(label: "Enum.map process output")
+    |> Enum.join()
+    |> IO.inspect(label: "output of Enum.join")
+    |> patch()
```
Nice work on the piping

alanvardy commented on elixir/markdown/lib/markdown.ex:
```diff
@@ -11,37 +11,53 @@ defmodule Markdown do
-      "*" -> parse_list_md_level(t)
-      _ -> enclose_with_paragraph_tag(String.split(t))
+  #I removed the nested if structure and replaced it with 'case'"
+  # changed 't' into 'line' because it is the result of String.split in the parse function
```
:+1:

alanvardy commented on elixir/markdown/lib/markdown.ex:
```diff
@@ -11,37 +11,53 @@ defmodule Markdown do
   def parse(m) do
-    patch(Enum.join(Enum.map(String.split(m, "\n"), fn t -> process(t) end)))
-  end
+    m
```
Is there a better variable that could be used rather than `m`?

alanvardy replied:
No response to this question yet

CorneliaKelinske replied:
I have changed it to "markdown_text". I'm not 100% sure if this is a good choice. Maybe 'source_text'?

alanvardy replied:
I think either sounds good, maybe `markdown_text` could be better as it is a bit more specific?

alanvardy commented on elixir/markdown/lib/markdown.ex:
```diff
@@ -11,37 +11,53 @@ defmodule Markdown do
+    [h | t] = String.split(line)
     {to_string(String.length(h)), Enum.join(t, " ")}
+    |> IO.inspect(label: "output of parse_header_md_level")
+    # {"number", "text minus the header markup"}
```
Is this a forgotten comment?

One thing that might be able to improve the readability of line 41 is some well-named variables.

alanvardy replied:
?

CorneliaKelinske replied:
The forgotten comment is deleted and I have changed the variables.

alanvardy replied:
:+1:

alanvardy commented on elixir/markdown/lib/markdown.ex:
```diff
@@ -11,37 +11,53 @@ defmodule Markdown do
   end
 
   defp enclose_with_header_tag({hl, htl}) do
     "<h" <> hl <> ">" <> htl <> "</h" <> hl <> ">"
```
Is the htl variable just text? Can that variable name be improved?

alanvardy replied:
?

alanvardy reviewed (commented):
Looking good, the rampant `IO.inspects make it a bit harder to read and probably clutter up the terminal a bit?

Feel free to ping me for a review whenever :+1:

CorneliaKelinske commented:
Let me know if I can submit this to exercism now. I'm kinda done effing around with it.
Effing is not a bad word

alanvardy commented:
> Let me know if I can submit this to exercism now. I'm kinda done effing around with it.
> Effing is not a bad word

![2021-06-01_10-04](https://user-images.githubusercontent.com/38899847/120363231-bddfc400-c2c0-11eb-86fb-abd7666613c1.png)

CorneliaKelinske commented:
Thank you for the in-depth review. I have addressed all the comments and will upload the code with the latest changes now.
Please note: I have also added comments throughout the code that explain the changes I have made.
Let me know if there is anything else that sould be changed befor I merge.
Thank you!

alanvardy reviewed (approved):
Looks great! :star2:
